### ISSUE TITLE
k8s_info doesn't show "Route" resources

### ISSUE THREAD

**Issue by nathanegillett** [ansible-collections/community.okd#21]
##### SUMMARY
When using k8s_info module to get select resources under a namespace, none of kind "Route" are returned.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
k8s_info

##### ANSIBLE VERSION

```paste below
$ ansible --version
ansible 2.9.13
  config file = None
  configured module search path = ['/home/ngillett/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/ngillett/dev/python3/lib/python3.7/site-packages/ansible
  executable location = /home/ngillett/dev/python3/bin/ansible
  python version = 3.7.9 (default, Aug 20 2020, 14:07:20) [GCC 10.2.1 20200723 (Red Hat 10.2.1-1)]
```

##### CONFIGURATION

```paste below

```

##### OS / ENVIRONMENT

```
LSB Version:	:core-4.1-amd64:core-4.1-noarch
Distributor ID:	Fedora
Description:	Fedora release 32 (Thirty Two)
Release:	32
Codename:	ThirtyTwo
```

##### STEPS TO REPRODUCE

Create route resources in OpenShift, use k8s_info to fetch them.

```yaml
- name: Find existing resources
  k8s_info:
    api_key: "{{ exodus_gw_ocp_token }}"
    host: "{{ exodus_gw_ocp_host }}"
    namespace: "{{ exodus_gw_ocp_namespace }}"
    kind: "{{ item }}"
    ca_cert: "{{ exodus_gw_ca_cert }}"
  register: exodus_gw_resources
  with_items:
    - Pod
    - Service
    - ImageStream
    - DeploymentConfig
    - ReplicationController
    - NetworkPolicy
    - Route
```

##### EXPECTED RESULTS

Given that route resources exist under the namespace, they should be listed by the above.
```
$ oc get --namespace=exodus-gw-qa route
NAME                                             HOST/PORT                                   PATH      SERVICES    PORT      TERMINATION   WILDCARD
exodus-gw-qa.apps.ocp.prod.psi.redhat.com-host   exodus-gw-qa.apps.ocp.prod.psi.redhat.com             exodus-gw   http      edge          None
exodus-gw.qa.psi.redhat.com-cname                exodus-gw.qa.psi.redhat.com                           exodus-gw   http      edge          None
```

##### ACTUAL RESULTS

```paste below
ok: [exodus-gw-qa-project] => (item=Route) => {
    "ansible_loop_var": "item",
    "changed": false,
    "invocation": {
        "module_args": {
            "api_key": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "api_version": "v1",
            "ca_cert": "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem",
            "client_cert": null,
            "client_key": null,
            "context": null,
            "field_selectors": [],
            "host": "https://api.ocp.prod.psi.redhat.com:6443",
            "kind": "Route",
            "kubeconfig": null,
            "label_selectors": [],
            "name": null,
            "namespace": "exodus-gw-qa",
            "password": null,
            "proxy": null,
            "username": null,
            "validate_certs": null
        }
    },
    "item": "Route",
    "resources": []
}
```

**Comment by geerlingguy** [comment]
@tima - I am so confused... this issue magically moved over into this repo! (I didn't know that was even possible.)

**Comment by tima** [comment]
@geerlingguy ¯\_(ツ)_/¯

There was a "Transfer issue" action in my sidebar of the issue that I clicked. New features? My special powers on these two repos? (kubernetes and okd collections where my only options)

**Comment by nathanegillett** [comment]
@tima, @geerlingguy, thank you! I'll keep an eye on that V0.1 project for my team.

**Comment by fabianvf** [comment]
@nathanegillett what version of openshift are you using?

**Comment by nathanegillett** [comment]
@fabianvf We're using 4.3, from what I can tell. Not sure how to get the full version. The web console isn't much help.

**Comment by fabianvf** [comment]
I ran this playbook against a 3.11 and 4.5 OpenShift cluster

```
- hosts: localhost
  connection: local
  gather_facts: no

  vars:
    names: '{{ result.resources | map(attribute="metadata") | map(attribute="name") | list }}'

  tasks:
    - k8s_info:
        api_version: route.openshift.io/v1
        kind: Route
        namespace: default
      ignore_errors: yes
      register: result

    - debug: var=names

    - k8s_info:
        api_version: v1
        kind: Route
        namespace: default
      ignore_errors: yes
      register: result

    - debug: var=names

    - k8s_info:
        kind: Route
        namespace: default
      ignore_errors: yes
      register: result

    - debug: var=names
```

and got the results

```

PLAY [localhost] ******************************************************************************************************

TASK [k8s_info] *******************************************************************************************************
ok: [localhost]

TASK [debug] **********************************************************************************************************
ok: [localhost] => {
    "names": [
        "docker-registry",
        "registry-console"
    ]
}

TASK [k8s_info] *******************************************************************************************************
ok: [localhost]

TASK [debug] **********************************************************************************************************
ok: [localhost] => {
    "names": [
        "docker-registry",
        "registry-console"
    ]
}

TASK [k8s_info] *******************************************************************************************************
ok: [localhost]

TASK [debug] **********************************************************************************************************
ok: [localhost] => {
    "names": [
        "docker-registry",
        "registry-console"
    ]
}

PLAY RECAP ************************************************************************************************************
localhost                  : ok=6    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
```

Would you mind running the above playbook and seeing if your results differ significantly? (and if they do could you attach a paste with the `-vvv` output?)

**Comment by nathanegillett** [comment]
@fabianvf, only the first variant returned anything, which unblocks my current work.
Here's the output:

> TASK [exodus-gw : k8s_info] ****************************************************
task path: /home/ngillett/.ansible/roles/exodus-gw/tasks/clean.yml:23
<exodus-gw-qa-project> ESTABLISH LOCAL CONNECTION FOR USER: ngillett
<exodus-gw-qa-project> EXEC /bin/sh -c 'echo ~ngillett && sleep 0'
<exodus-gw-qa-project> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/ngillett/.ansible/tmp `"&& mkdir "` echo /home/ngillett/.ansible/tmp/ansible-tmp-1599591397.9637318-75051-38460253072233 `" && echo ansible-tmp-1599591397.9637318-75051-38460253072233="` echo /home/ngillett/.ansible/tmp/ansible-tmp-1599591397.9637318-75051-38460253072233 `" ) && sleep 0'
Using module file /home/ngillett/dev/python3/lib/python3.7/site-packages/ansible/modules/clustering/k8s/k8s_info.py
<exodus-gw-qa-project> PUT /home/ngillett/.ansible/tmp/ansible-local-74933_j40ngkt/tmpjt_gt_eo TO /home/ngillett/.ansible/tmp/ansible-tmp-1599591397.9637318-75051-38460253072233/AnsiballZ_k8s_info.py
<exodus-gw-qa-project> EXEC /bin/sh -c 'chmod u+x /home/ngillett/.ansible/tmp/ansible-tmp-1599591397.9637318-75051-38460253072233/ /home/ngillett/.ansible/tmp/ansible-tmp-1599591397.9637318-75051-38460253072233/AnsiballZ_k8s_info.py && sleep 0'
<exodus-gw-qa-project> EXEC /bin/sh -c '~/dev/python3/bin/python3.7 /home/ngillett/.ansible/tmp/ansible-tmp-1599591397.9637318-75051-38460253072233/AnsiballZ_k8s_info.py && sleep 0'
<exodus-gw-qa-project> EXEC /bin/sh -c 'rm -f -r /home/ngillett/.ansible/tmp/ansible-tmp-1599591397.9637318-75051-38460253072233/ > /dev/null 2>&1 && sleep 0'
ok: [exodus-gw-qa-project] => {
    "changed": false,
    "invocation": {
        "module_args": {
            "api_key": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "api_version": "route.openshift.io/v1",
            "ca_cert": "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem",
            "client_cert": null,
            "client_key": null,
            "context": null,
            "field_selectors": [],
            "host": "https://api.ocp.prod.psi.redhat.com:6443",
            "kind": "Route",
            "kubeconfig": null,
            "label_selectors": [],
            "name": null,
            "namespace": "exodus-gw-qa",
            "password": null,
            "proxy": null,
            "username": null,
            "validate_certs": null
        }
    },
    "resources": [
        {
            "apiVersion": "route.openshift.io/v1",
            "kind": "Route",
            "metadata": {
                "annotations": {
                    "haproxy.router.openshift.io/timeout": "10m"
                },
                "creationTimestamp": "2020-09-02T19:41:10Z",
                "labels": {
                    "app": "exodus-gw"
                },
                "name": "exodus-gw-qa.apps.ocp.prod.psi.redhat.com-host",
                "namespace": "exodus-gw-qa",
                "resourceVersion": "198808521",
                "selfLink": "/apis/route.openshift.io/v1/namespaces/exodus-gw-qa/routes/exodus-gw-qa.apps.ocp.prod.psi.redhat.com-host",
                "uid": "4b11c215-4c94-4949-806b-dd00c14514cc"
            },
            "spec": {
                "host": "exodus-gw-qa.apps.ocp.prod.psi.redhat.com",
                "port": {
                    "targetPort": "http"
                },
                "tls": {
                    "termination": "edge"
                },
                "to": {
                    "kind": "Service",
                    "name": "exodus-gw",
                    "weight": 100
                },
                "wildcardPolicy": "None"
            },
            "status": {
                "ingress": [
                    {
                        "conditions": [
                            {
                                "lastTransitionTime": "2020-09-02T19:41:10Z",
                                "status": "True",
                                "type": "Admitted"
                            }
                        ],
                        "host": "exodus-gw-qa.apps.ocp.prod.psi.redhat.com",
                        "routerCanonicalHostname": "apps.ocp.prod.psi.redhat.com",
                        "routerName": "default",
                        "wildcardPolicy": "None"
                    }
                ]
            }
        },
        {
            "apiVersion": "route.openshift.io/v1",
            "kind": "Route",
            "metadata": {
                "annotations": {
                    "haproxy.router.openshift.io/timeout": "10m"
                },
                "creationTimestamp": "2020-09-02T19:41:11Z",
                "labels": {
                    "app": "exodus-gw"
                },
                "name": "exodus-gw.qa.psi.redhat.com-cname",
                "namespace": "exodus-gw-qa",
                "resourceVersion": "198808560",
                "selfLink": "/apis/route.openshift.io/v1/namespaces/exodus-gw-qa/routes/exodus-gw.qa.psi.redhat.com-cname",
                "uid": "7eee9a9b-59db-4e72-847d-910dd597cbb0"
            },
            "spec": {
                "host": "exodus-gw.qa.psi.redhat.com",
                "port": {
                    "targetPort": "http"
                },
                "tls": {
                    "certificate": "---",
                    "key": "---",
                    "termination": "edge"
                },
                "to": {
                    "kind": "Service",
                    "name": "exodus-gw",
                    "weight": 100
                },
                "wildcardPolicy": "None"
            },
            "status": {
                "ingress": [
                    {
                        "conditions": [
                            {
                                "lastTransitionTime": "2020-09-02T19:41:11Z",
                                "status": "True",
                                "type": "Admitted"
                            }
                        ],
                        "host": "exodus-gw.qa.psi.redhat.com",
                        "routerCanonicalHostname": "apps.ocp.prod.psi.redhat.com",
                        "routerName": "default",
                        "wildcardPolicy": "None"
                    }
                ]
            }
        }
    ]
}
Read vars_file '../../vaults/qa/secret-vars.yml'

TASK [exodus-gw : debug] *******************************************************
task path: /home/ngillett/.ansible/roles/exodus-gw/tasks/clean.yml:33
ok: [exodus-gw-qa-project] => {
    "result.resources | map(attribute=\"metadata\") | map(attribute=\"name\") | list": [
        "exodus-gw-qa.apps.ocp.prod.psi.redhat.com-host",
        "exodus-gw.qa.psi.redhat.com-cname"
    ]
}
Read vars_file '../../vaults/qa/secret-vars.yml'

TASK [exodus-gw : k8s_info] ****************************************************
task path: /home/ngillett/.ansible/roles/exodus-gw/tasks/clean.yml:36
<exodus-gw-qa-project> ESTABLISH LOCAL CONNECTION FOR USER: ngillett
<exodus-gw-qa-project> EXEC /bin/sh -c 'echo ~ngillett && sleep 0'
<exodus-gw-qa-project> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/ngillett/.ansible/tmp `"&& mkdir "` echo /home/ngillett/.ansible/tmp/ansible-tmp-1599591399.223155-75073-250189848117666 `" && echo ansible-tmp-1599591399.223155-75073-250189848117666="` echo /home/ngillett/.ansible/tmp/ansible-tmp-1599591399.223155-75073-250189848117666 `" ) && sleep 0'
Using module file /home/ngillett/dev/python3/lib/python3.7/site-packages/ansible/modules/clustering/k8s/k8s_info.py
<exodus-gw-qa-project> PUT /home/ngillett/.ansible/tmp/ansible-local-74933_j40ngkt/tmplp9kydhq TO /home/ngillett/.ansible/tmp/ansible-tmp-1599591399.223155-75073-250189848117666/AnsiballZ_k8s_info.py
<exodus-gw-qa-project> EXEC /bin/sh -c 'chmod u+x /home/ngillett/.ansible/tmp/ansible-tmp-1599591399.223155-75073-250189848117666/ /home/ngillett/.ansible/tmp/ansible-tmp-1599591399.223155-75073-250189848117666/AnsiballZ_k8s_info.py && sleep 0'
<exodus-gw-qa-project> EXEC /bin/sh -c '~/dev/python3/bin/python3.7 /home/ngillett/.ansible/tmp/ansible-tmp-1599591399.223155-75073-250189848117666/AnsiballZ_k8s_info.py && sleep 0'
<exodus-gw-qa-project> EXEC /bin/sh -c 'rm -f -r /home/ngillett/.ansible/tmp/ansible-tmp-1599591399.223155-75073-250189848117666/ > /dev/null 2>&1 && sleep 0'
ok: [exodus-gw-qa-project] => {
    "changed": false,
    "invocation": {
        "module_args": {
            "api_key": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "api_version": "v1",
            "ca_cert": "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem",
            "client_cert": null,
            "client_key": null,
            "context": null,
            "field_selectors": [],
            "host": "https://api.ocp.prod.psi.redhat.com:6443",
            "kind": "Route",
            "kubeconfig": null,
            "label_selectors": [],
            "name": null,
            "namespace": "exodus-gw-qa",
            "password": null,
            "proxy": null,
            "username": null,
            "validate_certs": null
        }
    },
    "resources": []
}
Read vars_file '../../vaults/qa/secret-vars.yml'

TASK [exodus-gw : debug] *******************************************************
task path: /home/ngillett/.ansible/roles/exodus-gw/tasks/clean.yml:46
ok: [exodus-gw-qa-project] => {
    "result.resources | map(attribute=\"metadata\") | map(attribute=\"name\") | list": []
}
Read vars_file '../../vaults/qa/secret-vars.yml'

TASK [exodus-gw : k8s_info] ****************************************************
task path: /home/ngillett/.ansible/roles/exodus-gw/tasks/clean.yml:50
<exodus-gw-qa-project> ESTABLISH LOCAL CONNECTION FOR USER: ngillett
<exodus-gw-qa-project> EXEC /bin/sh -c 'echo ~ngillett && sleep 0'
<exodus-gw-qa-project> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/ngillett/.ansible/tmp `"&& mkdir "` echo /home/ngillett/.ansible/tmp/ansible-tmp-1599591410.5578256-75111-36736506572640 `" && echo ansible-tmp-1599591410.5578256-75111-36736506572640="` echo /home/ngillett/.ansible/tmp/ansible-tmp-1599591410.5578256-75111-36736506572640 `" ) && sleep 0'
Using module file /home/ngillett/dev/python3/lib/python3.7/site-packages/ansible/modules/clustering/k8s/k8s_info.py
<exodus-gw-qa-project> PUT /home/ngillett/.ansible/tmp/ansible-local-74933_j40ngkt/tmpvzzzgf0c TO /home/ngillett/.ansible/tmp/ansible-tmp-1599591410.5578256-75111-36736506572640/AnsiballZ_k8s_info.py
<exodus-gw-qa-project> EXEC /bin/sh -c 'chmod u+x /home/ngillett/.ansible/tmp/ansible-tmp-1599591410.5578256-75111-36736506572640/ /home/ngillett/.ansible/tmp/ansible-tmp-1599591410.5578256-75111-36736506572640/AnsiballZ_k8s_info.py && sleep 0'
<exodus-gw-qa-project> EXEC /bin/sh -c '~/dev/python3/bin/python3.7 /home/ngillett/.ansible/tmp/ansible-tmp-1599591410.5578256-75111-36736506572640/AnsiballZ_k8s_info.py && sleep 0'
<exodus-gw-qa-project> EXEC /bin/sh -c 'rm -f -r /home/ngillett/.ansible/tmp/ansible-tmp-1599591410.5578256-75111-36736506572640/ > /dev/null 2>&1 && sleep 0'
ok: [exodus-gw-qa-project] => {
    "changed": false,
    "invocation": {
        "module_args": {
            "api_key": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "api_version": "v1",
            "ca_cert": "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem",
            "client_cert": null,
            "client_key": null,
            "context": null,
            "field_selectors": [],
            "host": "https://api.ocp.prod.psi.redhat.com:6443",
            "kind": "Route",
            "kubeconfig": null,
            "label_selectors": [],
            "name": null,
            "namespace": "exodus-gw-qa",
            "password": null,
            "proxy": null,
            "username": null,
            "validate_certs": null
        }
    },
    "resources": []
}
Read vars_file '../../vaults/qa/secret-vars.yml'

TASK [exodus-gw : debug] *******************************************************
task path: /home/ngillett/.ansible/roles/exodus-gw/tasks/clean.yml:59
ok: [exodus-gw-qa-project] => {
    "result.resources | map(attribute=\"metadata\") | map(attribute=\"name\") | list": []
}
META: ran handlers
META: ran handlers

PLAY RECAP *********************************************************************
exodus-gw-qa-project       : ok=7    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0

**Comment by nathanegillett** [comment]
Update from our new 4.5 cluster:
```
TASK [exodus-gw : Find existing resources] *************************************
task path: /home/ngillett/.ansible/roles/exodus-gw/tasks/clean.yml:6
<exodus-gw-dev-project> ESTABLISH LOCAL CONNECTION FOR USER: ngillett
<exodus-gw-dev-project> EXEC /bin/sh -c 'echo ~ngillett && sleep 0'
<exodus-gw-dev-project> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/ngillett/.ansible/tmp `"&& mkdir /home/ngillett/.ansible/tmp/ansible-tmp-1601396532.629496-230336-28746015957606 && echo ansible-tmp-1601396532.629496-230336-28746015957606="` echo /home/ngillett/.ansible/tmp/ansible-tmp-1601396532.629496-230336-28746015957606 `" ) && sleep 0'
Using module file /home/ngillett/.local/lib/python3.8/site-packages/ansible/modules/clustering/k8s/k8s_info.py
<exodus-gw-dev-project> PUT /home/ngillett/.ansible/tmp/ansible-local-230222ui1sphn9/tmpcbiuk533 TO /home/ngillett/.ansible/tmp/ansible-tmp-1601396532.629496-230336-28746015957606/AnsiballZ_k8s_info.py
<exodus-gw-dev-project> EXEC /bin/sh -c 'chmod u+x /home/ngillett/.ansible/tmp/ansible-tmp-1601396532.629496-230336-28746015957606/ /home/ngillett/.ansible/tmp/ansible-tmp-1601396532.629496-230336-28746015957606/AnsiballZ_k8s_info.py && sleep 0'
<exodus-gw-dev-project> EXEC /bin/sh -c '~/dev/python3/bin/python3.7 /home/ngillett/.ansible/tmp/ansible-tmp-1601396532.629496-230336-28746015957606/AnsiballZ_k8s_info.py && sleep 0'
<exodus-gw-dev-project> EXEC /bin/sh -c 'rm -f -r /home/ngillett/.ansible/tmp/ansible-tmp-1601396532.629496-230336-28746015957606/ > /dev/null 2>&1 && sleep 0'
ok: [exodus-gw-dev-project] => {
    "changed": false,
    "invocation": {
        "module_args": {
            "api_key": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "api_version": "route.openshift.io/v1",
            "ca_cert": "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem",
            "client_cert": null,
            "client_key": null,
            "context": null,
            "field_selectors": [],
            "host": "https://api.ocp4.prod.psi.redhat.com:6443",
            "kind": "Route",
            "kubeconfig": null,
            "label_selectors": [],
            "name": null,
            "namespace": "exodus-gw-dev",
            "password": null,
            "proxy": null,
            "username": null,
            "validate_certs": null
        }
    },
    "resources": [
        {
            "apiVersion": "route.openshift.io/v1",
            "kind": "Route",
            "metadata": {
                "annotations": {
                    "haproxy.router.openshift.io/timeout": "10m"
                },
                "creationTimestamp": "2020-09-29T16:02:23Z",
                "labels": {
                    "app": "exodus-gw"
                },
                "managedFields": [
                    {
                        "apiVersion": "route.openshift.io/v1",
                        "fieldsType": "FieldsV1",
                        "fieldsV1": {
                            "f:metadata": {
                                "f:annotations": {
                                    ".": {},
                                    "f:haproxy.router.openshift.io/timeout": {}
                                },
                                "f:labels": {
                                    ".": {},
                                    "f:app": {}
                                }
                            },
                            "f:spec": {
                                "f:host": {},
                                "f:port": {
                                    ".": {},
                                    "f:targetPort": {}
                                },
                                "f:tls": {
                                    ".": {},
                                    "f:termination": {}
                                },
                                "f:to": {
                                    "f:kind": {},
                                    "f:name": {},
                                    "f:weight": {}
                                },
                                "f:wildcardPolicy": {}
                            }
                        },
                        "manager": "OpenAPI-Generator",
                        "operation": "Update",
                        "time": "2020-09-29T16:02:23Z"
                    },
                    {
                        "apiVersion": "route.openshift.io/v1",
                        "fieldsType": "FieldsV1",
                        "fieldsV1": {
                            "f:status": {
                                "f:ingress": {}
                            }
                        },
                        "manager": "openshift-router",
                        "operation": "Update",
                        "time": "2020-09-29T16:02:24Z"
                    }
                ],
                "name": "exodus-gw",
                "namespace": "exodus-gw-dev",
                "resourceVersion": "22417686",
                "selfLink": "/apis/route.openshift.io/v1/namespaces/exodus-gw-dev/routes/exodus-gw",
                "uid": "0c9f65d3-4ae9-4e3f-a47a-165d1ad1d869"
            },
            "spec": {
                "host": "exodus-gw.dev.psi.redhat.com",
                "port": {
                    "targetPort": "https"
                },
                "tls": {
                    "termination": "passthrough"
                },
                "to": {
                    "kind": "Service",
                    "name": "exodus-gw",
                    "weight": 100
                },
                "wildcardPolicy": "None"
            },
            "status": {
                "ingress": [
                    {
                        "conditions": [
                            {
                                "lastTransitionTime": "2020-09-29T16:02:24Z",
                                "status": "True",
                                "type": "Admitted"
                            }
                        ],
                        "host": "exodus-gw.dev.psi.redhat.com",
                        "routerCanonicalHostname": "apps.ocp4.prod.psi.redhat.com",
                        "routerName": "default",
                        "wildcardPolicy": "None"
                    }
                ]
            }
        }
    ]
}
Read vars_file '../../vaults/dev/secret-vars.yml'

TASK [exodus-gw : debug] *******************************************************
task path: /home/ngillett/.ansible/roles/exodus-gw/tasks/clean.yml:16
ok: [exodus-gw-dev-project] => {
    "exodus_gw_resources": {
        "changed": false,
        "failed": false,
        "resources": [
            {
                "apiVersion": "route.openshift.io/v1",
                "kind": "Route",
                "metadata": {
                    "annotations": {
                        "haproxy.router.openshift.io/timeout": "10m"
                    },
                    "creationTimestamp": "2020-09-29T16:02:23Z",
                    "labels": {
                        "app": "exodus-gw"
                    },
                    "managedFields": [
                        {
                            "apiVersion": "route.openshift.io/v1",
                            "fieldsType": "FieldsV1",
                            "fieldsV1": {
                                "f:metadata": {
                                    "f:annotations": {
                                        ".": {},
                                        "f:haproxy.router.openshift.io/timeout": {}
                                    },
                                    "f:labels": {
                                        ".": {},
                                        "f:app": {}
                                    }
                                },
                                "f:spec": {
                                    "f:host": {},
                                    "f:port": {
                                        ".": {},
                                        "f:targetPort": {}
                                    },
                                    "f:tls": {
                                        ".": {},
                                        "f:termination": {}
                                    },
                                    "f:to": {
                                        "f:kind": {},
                                        "f:name": {},
                                        "f:weight": {}
                                    },
                                    "f:wildcardPolicy": {}
                                }
                            },
                            "manager": "OpenAPI-Generator",
                            "operation": "Update",
                            "time": "2020-09-29T16:02:23Z"
                        },
                        {
                            "apiVersion": "route.openshift.io/v1",
                            "fieldsType": "FieldsV1",
                            "fieldsV1": {
                                "f:status": {
                                    "f:ingress": {}
                                }
                            },
                            "manager": "openshift-router",
                            "operation": "Update",
                            "time": "2020-09-29T16:02:24Z"
                        }
                    ],
                    "name": "exodus-gw",
                    "namespace": "exodus-gw-dev",
                    "resourceVersion": "22417686",
                    "selfLink": "/apis/route.openshift.io/v1/namespaces/exodus-gw-dev/routes/exodus-gw",
                    "uid": "0c9f65d3-4ae9-4e3f-a47a-165d1ad1d869"
                },
                "spec": {
                    "host": "exodus-gw.dev.psi.redhat.com",
                    "port": {
                        "targetPort": "https"
                    },
                    "tls": {
                        "termination": "passthrough"
                    },
                    "to": {
                        "kind": "Service",
                        "name": "exodus-gw",
                        "weight": 100
                    },
                    "wildcardPolicy": "None"
                },
                "status": {
                    "ingress": [
                        {
                            "conditions": [
                                {
                                    "lastTransitionTime": "2020-09-29T16:02:24Z",
                                    "status": "True",
                                    "type": "Admitted"
                                }
                            ],
                            "host": "exodus-gw.dev.psi.redhat.com",
                            "routerCanonicalHostname": "apps.ocp4.prod.psi.redhat.com",
                            "routerName": "default",
                            "wildcardPolicy": "None"
                        }
                    ]
                }
            }
        ]
    }
}
Read vars_file '../../vaults/dev/secret-vars.yml'

TASK [exodus-gw : Find existing resources] *************************************
task path: /home/ngillett/.ansible/roles/exodus-gw/tasks/clean.yml:19
<exodus-gw-dev-project> ESTABLISH LOCAL CONNECTION FOR USER: ngillett
<exodus-gw-dev-project> EXEC /bin/sh -c 'echo ~ngillett && sleep 0'
<exodus-gw-dev-project> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/ngillett/.ansible/tmp `"&& mkdir /home/ngillett/.ansible/tmp/ansible-tmp-1601396533.5012302-230355-224457653561198 && echo ansible-tmp-1601396533.5012302-230355-224457653561198="` echo /home/ngillett/.ansible/tmp/ansible-tmp-1601396533.5012302-230355-224457653561198 `" ) && sleep 0'
Using module file /home/ngillett/.local/lib/python3.8/site-packages/ansible/modules/clustering/k8s/k8s_info.py
<exodus-gw-dev-project> PUT /home/ngillett/.ansible/tmp/ansible-local-230222ui1sphn9/tmpuagqodo9 TO /home/ngillett/.ansible/tmp/ansible-tmp-1601396533.5012302-230355-224457653561198/AnsiballZ_k8s_info.py
<exodus-gw-dev-project> EXEC /bin/sh -c 'chmod u+x /home/ngillett/.ansible/tmp/ansible-tmp-1601396533.5012302-230355-224457653561198/ /home/ngillett/.ansible/tmp/ansible-tmp-1601396533.5012302-230355-224457653561198/AnsiballZ_k8s_info.py && sleep 0'
<exodus-gw-dev-project> EXEC /bin/sh -c '~/dev/python3/bin/python3.7 /home/ngillett/.ansible/tmp/ansible-tmp-1601396533.5012302-230355-224457653561198/AnsiballZ_k8s_info.py && sleep 0'
<exodus-gw-dev-project> EXEC /bin/sh -c 'rm -f -r /home/ngillett/.ansible/tmp/ansible-tmp-1601396533.5012302-230355-224457653561198/ > /dev/null 2>&1 && sleep 0'
ok: [exodus-gw-dev-project] => {
    "changed": false,
    "invocation": {
        "module_args": {
            "api_key": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "api_version": "v1",
            "ca_cert": "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem",
            "client_cert": null,
            "client_key": null,
            "context": null,
            "field_selectors": [],
            "host": "https://api.ocp4.prod.psi.redhat.com:6443",
            "kind": "Route",
            "kubeconfig": null,
            "label_selectors": [],
            "name": null,
            "namespace": "exodus-gw-dev",
            "password": null,
            "proxy": null,
            "username": null,
            "validate_certs": null
        }
    },
    "resources": [
        {
            "apiVersion": "route.openshift.io/v1",
            "kind": "Route",
            "metadata": {
                "annotations": {
                    "haproxy.router.openshift.io/timeout": "10m"
                },
                "creationTimestamp": "2020-09-29T16:02:23Z",
                "labels": {
                    "app": "exodus-gw"
                },
                "managedFields": [
                    {
                        "apiVersion": "route.openshift.io/v1",
                        "fieldsType": "FieldsV1",
                        "fieldsV1": {
                            "f:metadata": {
                                "f:annotations": {
                                    ".": {},
                                    "f:haproxy.router.openshift.io/timeout": {}
                                },
                                "f:labels": {
                                    ".": {},
                                    "f:app": {}
                                }
                            },
                            "f:spec": {
                                "f:host": {},
                                "f:port": {
                                    ".": {},
                                    "f:targetPort": {}
                                },
                                "f:tls": {
                                    ".": {},
                                    "f:termination": {}
                                },
                                "f:to": {
                                    "f:kind": {},
                                    "f:name": {},
                                    "f:weight": {}
                                },
                                "f:wildcardPolicy": {}
                            }
                        },
                        "manager": "OpenAPI-Generator",
                        "operation": "Update",
                        "time": "2020-09-29T16:02:23Z"
                    },
                    {
                        "apiVersion": "route.openshift.io/v1",
                        "fieldsType": "FieldsV1",
                        "fieldsV1": {
                            "f:status": {
                                "f:ingress": {}
                            }
                        },
                        "manager": "openshift-router",
                        "operation": "Update",
                        "time": "2020-09-29T16:02:24Z"
                    }
                ],
                "name": "exodus-gw",
                "namespace": "exodus-gw-dev",
                "resourceVersion": "22417686",
                "selfLink": "/apis/route.openshift.io/v1/namespaces/exodus-gw-dev/routes/exodus-gw",
                "uid": "0c9f65d3-4ae9-4e3f-a47a-165d1ad1d869"
            },
            "spec": {
                "host": "exodus-gw.dev.psi.redhat.com",
                "port": {
                    "targetPort": "https"
                },
                "tls": {
                    "termination": "passthrough"
                },
                "to": {
                    "kind": "Service",
                    "name": "exodus-gw",
                    "weight": 100
                },
                "wildcardPolicy": "None"
            },
            "status": {
                "ingress": [
                    {
                        "conditions": [
                            {
                                "lastTransitionTime": "2020-09-29T16:02:24Z",
                                "status": "True",
                                "type": "Admitted"
                            }
                        ],
                        "host": "exodus-gw.dev.psi.redhat.com",
                        "routerCanonicalHostname": "apps.ocp4.prod.psi.redhat.com",
                        "routerName": "default",
                        "wildcardPolicy": "None"
                    }
                ]
            }
        }
    ]
}
Read vars_file '../../vaults/dev/secret-vars.yml'

TASK [exodus-gw : debug] *******************************************************
task path: /home/ngillett/.ansible/roles/exodus-gw/tasks/clean.yml:29
ok: [exodus-gw-dev-project] => {
    "exodus_gw_resources": {
        "changed": false,
        "failed": false,
        "resources": [
            {
                "apiVersion": "route.openshift.io/v1",
                "kind": "Route",
                "metadata": {
                    "annotations": {
                        "haproxy.router.openshift.io/timeout": "10m"
                    },
                    "creationTimestamp": "2020-09-29T16:02:23Z",
                    "labels": {
                        "app": "exodus-gw"
                    },
                    "managedFields": [
                        {
                            "apiVersion": "route.openshift.io/v1",
                            "fieldsType": "FieldsV1",
                            "fieldsV1": {
                                "f:metadata": {
                                    "f:annotations": {
                                        ".": {},
                                        "f:haproxy.router.openshift.io/timeout": {}
                                    },
                                    "f:labels": {
                                        ".": {},
                                        "f:app": {}
                                    }
                                },
                                "f:spec": {
                                    "f:host": {},
                                    "f:port": {
                                        ".": {},
                                        "f:targetPort": {}
                                    },
                                    "f:tls": {
                                        ".": {},
                                        "f:termination": {}
                                    },
                                    "f:to": {
                                        "f:kind": {},
                                        "f:name": {},
                                        "f:weight": {}
                                    },
                                    "f:wildcardPolicy": {}
                                }
                            },
                            "manager": "OpenAPI-Generator",
                            "operation": "Update",
                            "time": "2020-09-29T16:02:23Z"
                        },
                        {
                            "apiVersion": "route.openshift.io/v1",
                            "fieldsType": "FieldsV1",
                            "fieldsV1": {
                                "f:status": {
                                    "f:ingress": {}
                                }
                            },
                            "manager": "openshift-router",
                            "operation": "Update",
                            "time": "2020-09-29T16:02:24Z"
                        }
                    ],
                    "name": "exodus-gw",
                    "namespace": "exodus-gw-dev",
                    "resourceVersion": "22417686",
                    "selfLink": "/apis/route.openshift.io/v1/namespaces/exodus-gw-dev/routes/exodus-gw",
                    "uid": "0c9f65d3-4ae9-4e3f-a47a-165d1ad1d869"
                },
                "spec": {
                    "host": "exodus-gw.dev.psi.redhat.com",
                    "port": {
                        "targetPort": "https"
                    },
                    "tls": {
                        "termination": "passthrough"
                    },
                    "to": {
                        "kind": "Service",
                        "name": "exodus-gw",
                        "weight": 100
                    },
                    "wildcardPolicy": "None"
                },
                "status": {
                    "ingress": [
                        {
                            "conditions": [
                                {
                                    "lastTransitionTime": "2020-09-29T16:02:24Z",
                                    "status": "True",
                                    "type": "Admitted"
                                }
                            ],
                            "host": "exodus-gw.dev.psi.redhat.com",
                            "routerCanonicalHostname": "apps.ocp4.prod.psi.redhat.com",
                            "routerName": "default",
                            "wildcardPolicy": "None"
                        }
                    ]
                }
            }
        ]
    }
}
Read vars_file '../../vaults/dev/secret-vars.yml'

TASK [exodus-gw : Find existing resources] *************************************
task path: /home/ngillett/.ansible/roles/exodus-gw/tasks/clean.yml:32
<exodus-gw-dev-project> ESTABLISH LOCAL CONNECTION FOR USER: ngillett
<exodus-gw-dev-project> EXEC /bin/sh -c 'echo ~ngillett && sleep 0'
<exodus-gw-dev-project> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/ngillett/.ansible/tmp `"&& mkdir /home/ngillett/.ansible/tmp/ansible-tmp-1601396535.4446142-230374-166371673089324 && echo ansible-tmp-1601396535.4446142-230374-166371673089324="` echo /home/ngillett/.ansible/tmp/ansible-tmp-1601396535.4446142-230374-166371673089324 `" ) && sleep 0'
Using module file /home/ngillett/.local/lib/python3.8/site-packages/ansible/modules/clustering/k8s/k8s_info.py
<exodus-gw-dev-project> PUT /home/ngillett/.ansible/tmp/ansible-local-230222ui1sphn9/tmp89cn0dfx TO /home/ngillett/.ansible/tmp/ansible-tmp-1601396535.4446142-230374-166371673089324/AnsiballZ_k8s_info.py
<exodus-gw-dev-project> EXEC /bin/sh -c 'chmod u+x /home/ngillett/.ansible/tmp/ansible-tmp-1601396535.4446142-230374-166371673089324/ /home/ngillett/.ansible/tmp/ansible-tmp-1601396535.4446142-230374-166371673089324/AnsiballZ_k8s_info.py && sleep 0'
<exodus-gw-dev-project> EXEC /bin/sh -c '~/dev/python3/bin/python3.7 /home/ngillett/.ansible/tmp/ansible-tmp-1601396535.4446142-230374-166371673089324/AnsiballZ_k8s_info.py && sleep 0'
<exodus-gw-dev-project> EXEC /bin/sh -c 'rm -f -r /home/ngillett/.ansible/tmp/ansible-tmp-1601396535.4446142-230374-166371673089324/ > /dev/null 2>&1 && sleep 0'
ok: [exodus-gw-dev-project] => {
    "changed": false,
    "invocation": {
        "module_args": {
            "api_key": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "api_version": "v1",
            "ca_cert": "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem",
            "client_cert": null,
            "client_key": null,
            "context": null,
            "field_selectors": [],
            "host": "https://api.ocp4.prod.psi.redhat.com:6443",
            "kind": "Route",
            "kubeconfig": null,
            "label_selectors": [],
            "name": null,
            "namespace": "exodus-gw-dev",
            "password": null,
            "proxy": null,
            "username": null,
            "validate_certs": null
        }
    },
    "resources": [
        {
            "apiVersion": "route.openshift.io/v1",
            "kind": "Route",
            "metadata": {
                "annotations": {
                    "haproxy.router.openshift.io/timeout": "10m"
                },
                "creationTimestamp": "2020-09-29T16:02:23Z",
                "labels": {
                    "app": "exodus-gw"
                },
                "managedFields": [
                    {
                        "apiVersion": "route.openshift.io/v1",
                        "fieldsType": "FieldsV1",
                        "fieldsV1": {
                            "f:metadata": {
                                "f:annotations": {
                                    ".": {},
                                    "f:haproxy.router.openshift.io/timeout": {}
                                },
                                "f:labels": {
                                    ".": {},
                                    "f:app": {}
                                }
                            },
                            "f:spec": {
                                "f:host": {},
                                "f:port": {
                                    ".": {},
                                    "f:targetPort": {}
                                },
                                "f:tls": {
                                    ".": {},
                                    "f:termination": {}
                                },
                                "f:to": {
                                    "f:kind": {},
                                    "f:name": {},
                                    "f:weight": {}
                                },
                                "f:wildcardPolicy": {}
                            }
                        },
                        "manager": "OpenAPI-Generator",
                        "operation": "Update",
                        "time": "2020-09-29T16:02:23Z"
                    },
                    {
                        "apiVersion": "route.openshift.io/v1",
                        "fieldsType": "FieldsV1",
                        "fieldsV1": {
                            "f:status": {
                                "f:ingress": {}
                            }
                        },
                        "manager": "openshift-router",
                        "operation": "Update",
                        "time": "2020-09-29T16:02:24Z"
                    }
                ],
                "name": "exodus-gw",
                "namespace": "exodus-gw-dev",
                "resourceVersion": "22417686",
                "selfLink": "/apis/route.openshift.io/v1/namespaces/exodus-gw-dev/routes/exodus-gw",
                "uid": "0c9f65d3-4ae9-4e3f-a47a-165d1ad1d869"
            },
            "spec": {
                "host": "exodus-gw.dev.psi.redhat.com",
                "port": {
                    "targetPort": "https"
                },
                "tls": {
                    "termination": "passthrough"
                },
                "to": {
                    "kind": "Service",
                    "name": "exodus-gw",
                    "weight": 100
                },
                "wildcardPolicy": "None"
            },
            "status": {
                "ingress": [
                    {
                        "conditions": [
                            {
                                "lastTransitionTime": "2020-09-29T16:02:24Z",
                                "status": "True",
                                "type": "Admitted"
                            }
                        ],
                        "host": "exodus-gw.dev.psi.redhat.com",
                        "routerCanonicalHostname": "apps.ocp4.prod.psi.redhat.com",
                        "routerName": "default",
                        "wildcardPolicy": "None"
                    }
                ]
            }
        }
    ]
}
Read vars_file '../../vaults/dev/secret-vars.yml'

TASK [exodus-gw : debug] *******************************************************
task path: /home/ngillett/.ansible/roles/exodus-gw/tasks/clean.yml:41
ok: [exodus-gw-dev-project] => {
    "exodus_gw_resources": {
        "changed": false,
        "failed": false,
        "resources": [
            {
                "apiVersion": "route.openshift.io/v1",
                "kind": "Route",
                "metadata": {
                    "annotations": {
                        "haproxy.router.openshift.io/timeout": "10m"
                    },
                    "creationTimestamp": "2020-09-29T16:02:23Z",
                    "labels": {
                        "app": "exodus-gw"
                    },
                    "managedFields": [
                        {
                            "apiVersion": "route.openshift.io/v1",
                            "fieldsType": "FieldsV1",
                            "fieldsV1": {
                                "f:metadata": {
                                    "f:annotations": {
                                        ".": {},
                                        "f:haproxy.router.openshift.io/timeout": {}
                                    },
                                    "f:labels": {
                                        ".": {},
                                        "f:app": {}
                                    }
                                },
                                "f:spec": {
                                    "f:host": {},
                                    "f:port": {
                                        ".": {},
                                        "f:targetPort": {}
                                    },
                                    "f:tls": {
                                        ".": {},
                                        "f:termination": {}
                                    },
                                    "f:to": {
                                        "f:kind": {},
                                        "f:name": {},
                                        "f:weight": {}
                                    },
                                    "f:wildcardPolicy": {}
                                }
                            },
                            "manager": "OpenAPI-Generator",
                            "operation": "Update",
                            "time": "2020-09-29T16:02:23Z"
                        },
                        {
                            "apiVersion": "route.openshift.io/v1",
                            "fieldsType": "FieldsV1",
                            "fieldsV1": {
                                "f:status": {
                                    "f:ingress": {}
                                }
                            },
                            "manager": "openshift-router",
                            "operation": "Update",
                            "time": "2020-09-29T16:02:24Z"
                        }
                    ],
                    "name": "exodus-gw",
                    "namespace": "exodus-gw-dev",
                    "resourceVersion": "22417686",
                    "selfLink": "/apis/route.openshift.io/v1/namespaces/exodus-gw-dev/routes/exodus-gw",
                    "uid": "0c9f65d3-4ae9-4e3f-a47a-165d1ad1d869"
                },
                "spec": {
                    "host": "exodus-gw.dev.psi.redhat.com",
                    "port": {
                        "targetPort": "https"
                    },
                    "tls": {
                        "termination": "passthrough"
                    },
                    "to": {
                        "kind": "Service",
                        "name": "exodus-gw",
                        "weight": 100
                    },
                    "wildcardPolicy": "None"
                },
                "status": {
                    "ingress": [
                        {
                            "conditions": [
                                {
                                    "lastTransitionTime": "2020-09-29T16:02:24Z",
                                    "status": "True",
                                    "type": "Admitted"
                                }
                            ],
                            "host": "exodus-gw.dev.psi.redhat.com",
                            "routerCanonicalHostname": "apps.ocp4.prod.psi.redhat.com",
                            "routerName": "default",
                            "wildcardPolicy": "None"
                        }
                    ]
                }
            }
        ]
    }
}
META: ran handlers
META: ran handlers

PLAY RECAP *********************************************************************
exodus-gw-dev-project      : ok=7    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0 
```

**Comment by tima** [comment]
We believe this is a bug in that particular version of OpenShift. We couldn't reproduce it. Closing, but if it comes back up again please let us know in a new issue.
